### PR TITLE
Draft: add registry caching documentation

### PR DESCRIPTION
**Pull request by marians**
### What this PR does

- Add architecture page about registry in the management cluster
- Add tutorial on how to use that registry as a cache from a workload cluster

Context: https://github.com/giantswarm/roadmap/issues/3065

This will have to stay unmerged while team honeybadger has to figure out some details and improvements.

### CODE REVIEW

File: src/content/overview/architecture/registry/_index.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+owner:
+  - https://github.com/orgs/giantswarm/teams/team-honeybadger
+last_review_date: 2024-05-24
+---
+
+Outline
+
+- In general, any accessible container registry can be used with the Giant Swarm platform.
+- As of DATE, management clusters includes a container registry (Zot).
+- The registry in the management cluster is used as a pull-through cache from management cluster nodes.
```
Comment (pipo02mix) [inline src/content/overview/architecture/registry/_index.md:21]: I would explain briefly what means a pull-through cache

File: src/content/overview/architecture/registry/_index.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+- The registry in the management cluster can also be used from workload clusters.
```
Comment (pipo02mix) [inline src/content/overview/architecture/registry/_index.md:22]: I would just mention it and link to tutorial

**Review by pipo02mix — COMMENTED** [review]
Both outlines makes sense to me

**Comment by uvegla** [comment]
I intend https://github.com/giantswarm/docs/pull/2249 as a replacement for this.

**Pull request closed without merging**